### PR TITLE
Add species conversion

### PR DESCRIPTION
**Pull request by bryanwweber**
- [x] Tests added
- [x] Added entry into `CHANGELOG.md`

Changes proposed in this pull request:
 - Build stable/devel docs and upload separately
 - Add `species_conversion` keyword argument to convert species names

@pr-omethe-us/chemked

### CODE REVIEW

File: pyked/chemked.py [hunk 1]
```diff
@@ -640,35 +640,109 @@ def __init__(self, properties):
...
+
+        Arguments:
+            species_conversion (`dict`, optional): Mapping of species identifier to a
+                species name. This argument should be supplied when the name of the
+                species in the ChemKED YAML file does not match the name of the same
+                species in a chemical kinetic mechanism. The species identifier (the key
+                of the mapping) can be the name, InChI, or SMILES provided in the ChemKED
+                file, while the value associated with a key should be the desired name in
+                the Cantera format output string.
+
```
Comment (kyleniemeyer) [inline pyked/chemked.py]: could we include a simple example in the docstring here?

File: pyked/chemked.py [hunk 2]
```diff
@@ -640,35 +640,109 @@ def __init__(self, properties):
...
         Returns:
             `str`: String in the ``SPEC: AMT, SPEC: AMT`` format
```
Comment (kyleniemeyer) [inline pyked/chemked.py]: This is old, but Cantera actually wants `SPEC:AMT, SPEC:AMT, SPEC:AMT` (i.e., no spaces after the colons)

File: pyked/chemked.py [hunk 3]
```diff
@@ -640,35 +640,109 @@ def __init__(self, properties):
...
+            for c in self.composition:
+                amount = c['amount'].magnitude/factor
+                idents = [c.get(s) for s in ['species-name', 'InChI', 'SMILES'] if c.get(s, False)]
+                present = [i in species_conversion for i in idents]
+                if not any(present):
+                    comps.append('{!s}:{:.4e}'.format(c['species-name'], amount))
+                else:
+                    if len([i for i in present if i]) > 1:
+                        raise ValueError('More than one conversion present for species {}'.format(
+                                       c['species-name']))
```
Comment (kyleniemeyer) [inline pyked/chemked.py]: minor formatting issue, but can the second line align with the first, and have the final `)` on a new line, to match (what I think is) our style?

File: pyked/chemked.py [hunk 4]
```diff
@@ -640,35 +640,109 @@ def __init__(self, properties):
...
+            species_conversion (`dict`, optional): Mapping of species identifier to a
+                species name. This argument should be supplied when the name of the
+                species in the ChemKED YAML file does not match the name of the same
+                species in a chemical kinetic mechanism. The species identifier (the key
+                of the mapping) can be the name, InChI, or SMILES provided in the ChemKED
+                file, while the value associated with a key should be the desired name in
+                the Cantera format output string.
+
+        Returns:
+            `str`: String of mass fractions in the ``SPEC: AMT, SPEC: AMT`` format
```
Comment (kyleniemeyer) [inline pyked/chemked.py]: again, should be `SPEC:AMT, SPEC:AMT` (no spaces after colons)

File: pyked/chemked.py (hunk 2 above)
Comment (bryanwweber) [reply pyked/chemked.py]: Oh, I thought I fixed that...

File: pyked/chemked.py [hunk 5]
```diff
@@ -188,7 +188,7 @@ def get_dataframe(self, output_columns=None):
         app_index = valid_labels.index('apparatus')
         valid_labels[app_index:app_index + 1] = ['apparatus:' + a for a in Apparatus._fields]
 
-        species_list = list(set([s['species-name'] for d in self.datapoints for s in d.composition]))
+        species_list = list(set([s['species-name'] for d in self.datapoints for s in d.composition]))  # noqa: E501
```
[the remark this answers is not in the document]
Comment (bryanwweber) [reply pyked/chemked.py]: No quality assurance for E501 (line too long). The line is one character too long, but I think reformatting it to multiple lines would be harder to read.

[the remark this answers is not in the document]
Comment (kyleniemeyer) [reply pyked/chemked.py]: got it.

**Pull request merged**